lib/std/mem.zig

```diff
@@ -10,9 +10,19 @@ const testing = std.testing;
 const Endian = std.builtin.Endian;
 const native_endian = builtin.cpu.arch.endian();
 
+/// Runtime known minimum page size.
+/// If the page size cannot be determined at runtime,
+/// it will use the comptime known one.
+pub fn getPageSize() usize {
+    return switch (builtin.os.tag) {
+        .linux => if (builtin.link_libc) @intCast(std.c.sysconf(30)) else std.os.linux.getauxval(std.elf.AT_PAGESZ),
+        else => page_size,
```

Comment:
Kernels provide compilation and/or runtime configuration of the page size, so `page_size` is wrong. Feel free to use the bits from #16638.

Reply:
Alright, I'll add the bit when I am able to next.